File: codebases/backend/DTO/detail/php/FileUploadDetail.php
<?php

/**
 * Class FileUploadDetail
 * 개별 파일 업로드의 세부 정보를 정의하는 클래스
 */
class FileUploadDetail {
    private string $fileName;
    private string $status;
    private string $fileType;

    public function __construct(string $fileName='', string $status='', string $fileType='') {
        $this->fileName = $fileName;
        $this->status = $status;
        $this->fileType = $fileType;
    }

    public static function fromArray(array $data): self {
        return new self(
            $data['file_name'] ?? '',
            $data['file_handle_status'] ?? '',
            $data['file_type'] ?? ''
        );
    }

    public function toFileUploadArray(): array {
        return [
            'file_name' => $this->fileName,
            'file_handle_status' => $this->status,
            'file_type' => $this->fileType,
        ];
    }
}
?>

File: codebases/backend/DTO/detail/py/FileUploadDetail.py
class FileUploadDetail:
    """
    파일 업로드의 세부 정보를 저장하는 클래스
    """

    def __init__(self, status: str, file_type: str):
        self._status = status
        self._file_type = file_type

    @property
    def status(self):
        return self._status

    @status.setter
    def status(self, value):
        self._status = value

    @property
    def file_type(self):
        return self._file_type

    @file_type.setter
    def file_type(self, value):
        self._file_type = value

    def to_dict(self) -> dict:
        return {
            'status': self.status,
            'fileType': self.file_type,
        }
